src/server/campaignd/server.cpp

```diff
@@ -1047,6 +1047,12 @@ void server::handle_request_campaign_list(const server::request& req)
 		// or irrelevant to clients
 		j.remove_attributes("passphrase", "passhash", "passsalt", "upload_ip", "email");
 
+		// don't include icons if requested
+		// also skip DataURI options that are unreasonably large
+		if(!req.cfg["icons"].to_bool(true) || j["icon"].str().size() > 500'000) {
```

Comment:
Just `icons` is a bit vague. I prefer something clearer like `send_icons` or so and perhaps just overwrite data URIs. Also just silently ignoring big icons is confusing. The server should rather not accept addons with too big icons and the limit should be configurable.

Is there a visual difference between removing the icon and the blank png?

Reply:
What do you mean by "removing the icon" specifically?

Reply:
Remove the attribute instead of setting it to a different value.

Reply:
I'd say for overwriting data URIs it makes sense to use the blank png otherwise I'd remove the attribute and log no error.

Reply:
Updated.